### DISCUSSION TITLE
how to add the talk map to the talkpage

### DISCUSSION THREAD

**Discussion by arnaou** [academicpages/academicpages.github.io#2271] [Q&A]
Hi,

I am kinda new to this so i am not entirely sure on how to add the generated map to the page.

any hints?

thanks in advance

**Comment by rjzupkoii** [comment]
My understanding is that the `talkmap.ipynb` or `talkmapy.py` should setup everything needed, but I never used it for my own side and as you might be able to guess from the timestamps in the repository, it's also not very well supported at this point.

**Reply by arnaou** [reply]
tried that, didn't work :/ . but thanks for the reply :)

**Reply by dimitra-maoutsa** [reply]
I just found out how to set this up for my own site, so I'll leave here the instructions for reference.

In the `_config.yml `line 89 you set the  `talkmap_link` to `true`.

Then create a dictionary for github workflows: `.github/workflows/`  and create in there a file that set up a github action for running the jupyter notebook that scrapes the talk locations every time you push changes.

Call it sth like: `scrape_talks.yml`

and inside add the following: 



name: Scrape Talk Locations

on:
  push:
    paths:
      - 'talks/**'
      - 'talkmap.ipynb'

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2

    - name: Set up Python
      uses: actions/setup-python@v2
      with:
        python-version: '3.9'  # Specify the Python version you need

    - name: Install dependencies
      run: |
        pip install jupyter pandas requests beautifulsoup4 geopy  # Add other dependencies as needed
        pip install getorg --upgrade

    - name: Run Jupyter Notebook
      run: |
        jupyter nbconvert --to notebook --execute talkmap.ipynb --output talkmap_out.ipynb

    - name: Commit changes
      run: |
        git config user.name "github-actions[bot]"
        git config user.email "github-actions[bot]@users.noreply.github.com"
        git add .
        git commit -m "Automated update of talk locations" || echo "No changes to commit"
        git push
